### ISSUE TITLE
False positive on `ForbiddenMethodCall`

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#8282]
```kotlin
@Test
fun `should report property setters call 2`() {
    val code = """
        import java.util.Calendar
        
        fun main() {
            val calendar = Calendar.getInstance()
            val day = calendar.firstDayOfWeek
            calendar.firstDayOfWeek = 1
        }
    """.trimIndent()
    val findings = ForbiddenMethodCall(
        TestConfig(METHODS to listOf("java.util.Calendar.getFirstDayOfWeek", "java.util.Calendar.setFirstDayOfWeek"))
    ).lintWithContext(env, code)
    assertThat(findings).hasSize(2)
}
```

This test should pass but it fails because this code reports 4 findings instead of just 2. I know how to fix this but I'll wait to merge #8281 to avoid conflicts.
